testsuite/features/step_definitions/navigation_steps.rb

```diff
@@ -251,6 +255,20 @@
 #
 When(/^I click on "([^"]*)"$/) do |text|
   click_button_and_wait(text, match: :first)
+
+  # In case of a search reindex not having finished yet, keeps retrying until successful search or timeout
+  if text == 'Search' && has_text?('Could not connect to search server.', wait: 0)
+    click_button(text, match: :first, wait: false)
+    start = Time.new
+    timeout = 5
+    click_button(text, match: :first, wait: false) while
+      (has_text?('Could not connect to search server.', wait: 0) || has_text?('No matches found', wait: 0)) &&
+      (Time.new - start <= timeout)
```

Comment:
Nice. I wonder if we can have similar cases.
If we have a similar case in the future, we can try to remember this and extract this to a method.